crates/subspace-runtime-primitives/src/lib.rs

```diff
@@ -56,6 +56,24 @@ pub const DOMAINS_PRUNING_DEPTH_MULTIPLIER: u32 = 2;
 /// Domains Block pruning depth.
 pub const DOMAINS_BLOCK_PRUNING_DEPTH: u32 = 14_400;
 
+/// Additional Canonicalization number on top of Confirmation depth for state_db
+/// forced canonicalization of block held in memory
+/// Delay = confirmation_depth_k + addition_blocks
+/// Example:
+///     confirmation_depth_k = 100
+///     additional_blocks = 50
+///     delay = 100 + 50 = 150
+/// This will be number of blocks NonCanonicalOverlay will keep in the memory.
+/// Blocks beyond this value will be forced to canonicalize and written to state_db.
+/// Forcing does not mean we finalize the block but rather we prune all the sibling blocks
+/// at the height and canonicalize the best block.
+/// We use additional blocks even though consensus blocks are confirmed at the depth_k itself
+/// to be on safe side.
```

Comment:
I'm not sure what "safe side" means and why 50 is a good value for that. I'd set it to 0 instead (meaning removing this constant).